File: apps/cli/src/lib/clack/prompts/messages.ts
import type { Writable } from 'node:stream';
import color from 'picocolors';
import { type CommonOptions, S_BAR_END } from './common';

export const cancel = (message = '', opts?: CommonOptions) => {
	const output: Writable = opts?.output ?? process.stdout;
	output.write(`${color.gray(S_BAR_END)}  ${color.red(message)}\n\n`);
};

export const intro = (title = '', opts?: CommonOptions) => {
	const output: Writable = opts?.output ?? process.stdout;
	output.write(`\n${title}\n`);
};

export const outro = (message = '', opts?: CommonOptions) => {
	const output: Writable = opts?.output ?? process.stdout;
	output.write(`\n${message}\n\n`);
};

File: apps/cli-zig/src/lib/clack/prompts/messages.zig
const std = @import("std");
const color = @import("../../picocolors.zig");
const common = @import("common.zig");
const CommonOptions = common.CommonOptions;
const S_BAR_END = common.S_BAR_END;

//
// Writes a cancellation message.
//
pub fn cancel(allocator: std.mem.Allocator, io: std.Io, message: []const u8, opts: CommonOptions) !void {
    const output = common.resolveOutput(io, opts);
    try output.print("{s}  {s}\n\n", .{ try color.gray(allocator, S_BAR_END()), try color.red(allocator, message) });
    try output.flush();
}

//
// Writes an intro title.
//
pub fn intro(io: std.Io, title: []const u8, opts: CommonOptions) !void {
    const output = common.resolveOutput(io, opts);
    try output.print("\n{s}\n", .{title});
    try output.flush();
}

//
// Writes an outro message.
//
pub fn outro(io: std.Io, message: []const u8, opts: CommonOptions) !void {
    const output = common.resolveOutput(io, opts);
    try output.print("\n{s}\n", .{message});
    try output.flush();
}
